### PR TITLE
feat: support run in stream subsystem

### PR DESCRIPTION
**Pull request by tzssangglass**

### CODE REVIEW

**Comment by tzssangglass** [comment]
It looks like this repository doesn't have CI, do we need to add CI to this repository first? @spacewander

**Comment by spacewander** [comment]
> It looks like this repository doesn't have CI, do we need to add CI to this repository first? @spacewander

Yes. Let's submit another PR to convert travis CI to github Action.
For security concern, we can't enable travis CI.

**Comment by tzssangglass** [comment]
Then let this PR wait for a while

**Comment by tzssangglass** [comment]
PTAL @spacewander

**Comment by spacewander** [comment]
Is the CI failure expected?

**Comment by tzssangglass** [comment]
> Is the CI failure expected?

checked, yes

**Comment by spacewander** [comment]
It's strange. Why kong keeps test cases which are expected to fail?

**Comment by tzssangglass** [comment]
> It's strange. Why kong keeps test cases which are expected to fail?

The test case that failed in our github action ran successfully in their travis-ci. ref: https://app.travis-ci.com/github/Kong/lua-resty-dns-client/jobs/572291660

The difference between ours and theirs is the version number of OpenResty.

When I have time, I investigate why CIs fail.

**Comment by tzssangglass** [comment]
@spacewander It is really not possible to reproduce errors in CI locally.

It is related to these factors.

for failed cases in `spec/client_cache_spec.lua`, the following factors contributed to the failed
1. using 8.8.8.8 as the DNS server in the test case
2. the versions of OpenResty running in Travis-ci are 1.15.8.3 and 1.17.8.2
3. the network environment affects the running of TTL-related test cases

for failed cases in `spec/client_spec.lua`, the following factors contributed to the failed
1. the host `srvtest.thijsschreijer.nl` has been down, can't be resolver

for other failed, unstable appears.

**Comment by tzssangglass** [comment]
@spacewander annotated with some test cases ......

File: t/03-stream-phases.t [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        local answers, err = client.resolve(host, { qtype = typ })
+
+        if not answers then
+            ngx.say("failed to resolve: ", err)
+        end
+
+        ngx.print("address name: ", answers[1].name)
+    }
+
+    content_by_lua return;
```
Comment (spacewander) [inline t/03-stream-phases.t]: Better to use content_by_lua_block

Comment (tzssangglass) [reply t/03-stream-phases.t]: done

File: t/03-stream-phases.t [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+run_tests();
+
+__DATA__
+
+=== TEST 1: support preread phase in stream subsystem
+--- stream_config eval
+    "lua_package_path 'src/?.lua;;';"
+--- stream_server_config
+    content_by_lua_block {
```
Comment (spacewander) [inline t/03-stream-phases.t]: IMHO, this fix is adding support in the preread phase? Why do you test it in the content phase?

Comment (tzssangglass) [reply t/03-stream-phases.t]: my mistake, I misunderstood what https://github.com/api7/lua-resty-dns-client/pull/1#discussion_r1040505114 meant.

Comment (tzssangglass) [reply t/03-stream-phases.t]: done

**Review by spacewander — APPROVED** [review]

**Pull request merged**